### PR TITLE
BLD: updated requirements files

### LINKED ISSUE #6: BLD: Separate requirements.txt
Some requirements are not necessary to play whit this project. I suggest change it to something like this:

* `requirements-base.txt` : Contains the abstract libraries used in this project (without versions).
* `requirements-dev.txt` : Contains tools used by a developer (inside should have a reference to `requirements-test.txt` ?  ).
* `requirements-test.txt` : Contains libraries used for testing.
* `requirements.txt` : This is the usually version of a requirements, contains fixed versions to make this project run properly.

This can be useful to install only what we need.

### PR DESCRIPTION
**Pull request by emvalbuena**
Closes #6.

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+pandas
```
Comment (asdf8601) [inline requirements-dev.txt:1]: I expecting to see packages like flake8, sphinx, ... here.

File: requirements-test.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+-r requirements-dev.txt
```
Comment (asdf8601) [inline requirements-test.txt:1]: I recommend move this line to `req-dev` replacing dev by test, of course.

File: requirements.txt [hunk 3]
```diff
@@ -3,4 +3,4 @@ pipdeptree==0.13.2
 QuantStats==0.0.25
 tqdm==4.43.0
 wheel==0.34.2
-xlrd==1.2.0
+xlrd==1.2.0
```
Comment (asdf8601) [inline requirements.txt:6]: I expected to see here the same (or more) packages than `requirements-base` has,  with their versions.

File: requirements-dev.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+numpy
+matplotlib
+tqdm
+requests
+xlrd
+tabulate
+yfinance
```
Comment (asdf8601) [inline requirements-dev.txt:8]: we can add `-r requirements.txt`

**Review by asdf8601 — CHANGES_REQUESTED** [review]
I suggest some changes, let me know what do you think.

File: requirements-dev.txt (hunk 4 above)
Comment (emvalbuena) [reply requirements-dev.txt:8]: But then why do you want them separate?

Comment (asdf8601) [reply requirements-dev.txt:8]: * `requirements-dev.txt` for developers this means if I'm a developer and I do `pip install -r requirementes-dev.txt` I will have my environment ready to testing, runing and developing code on this project.
* Sometimes, if we use a CI/CD environment, we only want install testing tools that's why `requirements-test.txt` exists. 
* If we can check a common user installation we can use `requirement-base.txt`.

Comment (emvalbuena) [reply requirements-dev.txt:8]: I see. 

What's the difference then between `requirements.txt` and `requirements-base.txt`?

**Comment by mmngreco** [comment]
`requirements.txt` should be the concrete version (each package with their versions) of `requirements-base.txt` and ideally the `requiments.txt` defines the whole environment (100% reproducible) , this include all packages installed after checking that the library works properly (or all test had been passed).

File: requirements-dev.txt (hunk 4 above)
Comment (asdf8601) [reply requirements-dev.txt:8]: `requirements.txt` should be the concrete version (each package with their versions) of `requirements-base.txt` and ideally the `requiments.txt` defines the whole environment (100% reproducible) , this include all packages installed after checking that the library works properly (or all test had been passed).

**Comment by emvalbuena** [comment]
Closed due to cleaner and more advanced pull request coming in.

**Pull request closed without merging**